PyInstaller/building/api.py

```diff
@@ -171,8 +177,44 @@ def assemble(self):
                 self.code_dict[entry[0]] = self.__get_code(entry[0], entry[1])
         # sort content alphabetically to support reproducible builds
         toc.sort()
+
+        # Remove leading parts of paths in code objects
+        self.code_dict = {
+            key: self._strip_paths_in_code(code)
+            for key, code in self.code_dict.items()
+        }
+
         pyz = ZlibArchiveWriter(self.name, toc, code_dict=self.code_dict, cipher=self.cipher)
 
+    def _strip_paths_in_code(self, co):
```

Comment:
If we'd add another parameter `new_filename=None` we'd only need to calculate the new filename once.